packages/plugin-http-client/src/rest/AbstractResource.ts

```diff
@@ -0,0 +1,170 @@
+import { GenericError, Dependencies } from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    REST_CLIENT_BASE_API_URL: string;
+  }
+}
+
+import {
+  HttpClient,
+  HttpClientRequestMethod,
+  HttpClientRequestOptions,
+} from '../HttpClient';
+
+export class AbstractResource {
+  #httpClient: HttpClient;
+  #baseApiUrl: string;
+
+  static get $dependencies(): Dependencies {
+    return [HttpClient, '?REST_CLIENT_BASE_API_URL'];
```

Comment:
Why don't we take `REST_CLIENT_BASE_API_URL` from `$Settings`? I know it's been like this before, but this feels (and felt) like anti-pattern compared to how other plugins share settings.

Reply:
Moved to $Settings.